cpp/src/parquet/arrow/reader.cc

```diff
@@ -366,6 +351,30 @@ class ColumnChunkReaderImpl : public ColumnChunkReader {
   int row_group_index_;
 };
 
+struct RowGroupReader::Iterator : ::arrow::TableBatchReader {
+  explicit Iterator(const std::shared_ptr<Table>& table)
+      : TableBatchReader(*table), table_(table) {}
+  // TableBatchReader does not take ownership of table
+  std::shared_ptr<Table> table_;
+};
+
+Status RowGroupReader::MakeIterator(
+    std::unique_ptr<::arrow::Iterator<std::shared_ptr<::arrow::RecordBatch>>>* batches) {
```

Comment:
This could benefit from a `using RecordBatchIterator = ::arrow::Iterator<...>;`

Reply:
will do